src/shared/ask-password-api.c

```diff
@@ -1021,3 +1022,23 @@ int ask_password_auto(
 
         return -EUNATCH;
 }
+
+int acquire_user_ask_password_directory(char **ret) {
+
+        const char *e = secure_getenv("XDG_RUNTIME_DIR");
+        if (!e) {
+                if (ret)
+                        *ret = NULL;
+                return 0;
+        }
+
+        if (ret) {
+                _cleanup_free_ char *j = path_join(e, "systemd/ask-password");
+                if (!j)
+                        return -ENOMEM;
+
+                *ret = TAKE_PTR(j);
```

Comment:
xdg_user_runtime_dir

(I know it uses getenv rather than secure_ version, but there's a WIP PR from me that would port path-lookup.c to sd-path, hence should be fine)

Reply:
-> #34432